sympy/polys/matrices/sdm.py

```diff
@@ -680,6 +680,49 @@ def charpoly(A):
         """
         return A.to_ddm().charpoly()
 
+    def applyfunc(self, f, domain=None):
+        """Returns SDM after applying function f to each element
+
+        Examples
+        ========
+
+        >>> from sympy.polys.matrices.sdm import SDM
+        >>> from sympy import ZZ
+        >>> A = SDM.zeros((2, 2), ZZ)
+        >>> A.applyfunc(lambda x: x + ZZ(4))
+        {0: {0: 4, 1: 4}, 1: {0: 4, 1: 4}}
+
+        We can also specify the output :py:class:`~.Domain`
+
+        >>> from sympy import QQ
+        >>> A.applyfunc(lambda x: x + QQ(1, 2), QQ)
+        {0: {0: 1/2, 1: 1/2}, 1: {0: 1/2, 1: 1/2}}
+
+        """
+        if domain is None:
+            domain = self.domain
+
+        Mf = sdm_applyfunc(self, self.shape, f, domain)
+        return SDM(Mf, self.shape, domain)
+
+    def applyfunc_nonzero(self, f, domain=None):
+        """Returns SDM after applying function f to each non-zero element
```

Comment:
This docstring does not explain the purpose of this which is that it is efficient for sparse matrices.